spec.md

```diff
@@ -1617,6 +1626,40 @@ The CO MUST implement the specified error recovery behavior when it encounters t
 |-----------|-----------|-------------|-------------------|
 | Volume does not exist | 5 NOT_FOUND | Indicates that a volume corresponding to the specified `volume_id` does not exist. | Caller MUST verify that the `volume_id` is correct and that the volume is accessible and has not been deleted before retrying with exponential back off. |
 
+
+#### `ModifyVolume`
+
+A Controller plugin SHALL implement this RPC call. If the MODIFY_VOLUME capability is present, a Controller plugin MUST implement this RPC call. This RPC allows the CO to change mutable key attributes of a volume. 
+
+This operation MUST be idempotent. The preset parameters in the VolumePerformanceClass can be greater than or less than the current values. 
+
+
+```protobuf
+message ModifyVolumeRequest {
+  option (alpha_message) = true;
+
+  // Contains identity information for the existing volume.
+  // This field is REQUIRED.
+  string volume_id = 1;
+  // This field is OPTIONAL.This allows the CO to specify the
+  // mutable parameters to apply. 
+  map<string, string> mutable_parameters = 2;
+}
+
+message ModifyVolumeResponse {
+  option (alpha_message) = true;
+}
+
+```
+
+##### ModifyVolume Errors
+
+| Condition | gRPC Code | Description | Recovery Behavior |
+|-----------|-----------|-------------|-------------------|
+| Parameters not supported | 3 INVALID_ARGUMENT | Indicates that the CO has specified volume performance parameters not supported by the volume. | Caller MAY verify volume performance parameters. |
```

Comment:
"volume performance parameters" need to be updated.

Reply:
Agree, and we should be clear that SPs SHOULD report errors if keys that are not mutable are specified. I don't think we can outright declare that any unknown parameter is an error, because that will cause problems for forward/backward compatibility, but certainly keys that are known not to be valid should trigger and error.